quinn-proto/src/connection.rs

```diff
@@ -711,6 +716,7 @@ impl Connection {
     ) {
         self.remote_validated |= self.state.is_handshake() && space_id == SpaceId::Handshake;
         self.total_recvd = self.total_recvd.wrapping_add(size as u64);
+        self.reset_keep_alive(now);
```

Comment:
Just for my understanding, how do the triggers for `reset_keep_alive` and `reset_idle_timeout` different?

Reply:
`reset_keep_alive` is triggered on packet receive or *any* packet send; `reset_idle_timeout` is triggered on packet receive or on the first send following a receive, but not on any following sends. In other words, `rest_idle_timeout` is not triggered by automatic retransmissions, or even long outgoing flights of new data on a dead link.

Reply:
Makes sense, thanks!